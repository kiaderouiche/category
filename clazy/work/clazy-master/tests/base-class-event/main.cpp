#include <QtCore/QObject>

class MyObj : public QObject
{
public:

    bool event(QEvent *ev) override;
};

bool MyObj::event(QEvent *ev)
{
    if (false) {}

    return false; // Warning
}


class MyObj2 : public QObject
{
public:

    bool event(QEvent *ev) override
    {
        return false; // Warning
    }
};



class MyObj3 : public QObject
{
public:

    bool event2()
    {
        return false; // OK
    }

    bool event(QEvent *ev) override
    {
        return true; // OK
    }

    bool eventFilter(QObject *, QEvent *) override
    {
        return false; // OK
    }
};

class MyObj4 : public MyObj3
{
public:

    bool event2()
    {
        return false; // OK
    }

    bool event(QEvent *ev) override
    {
        return false; // Warning
    }

    bool eventFilter(QObject *, QEvent *) override
    {
        return false; // Warning
    }
};
