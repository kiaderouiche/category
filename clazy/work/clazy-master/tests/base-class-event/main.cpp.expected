base-class-event/main.cpp:14:5: warning: Return QObject::event() instead of false [-Wclazy-base-class-event]
base-class-event/main.cpp:24:9: warning: Return QObject::event() instead of false [-Wclazy-base-class-event]
base-class-event/main.cpp:61:9: warning: Return MyObj3::event() instead of false [-Wclazy-base-class-event]
base-class-event/main.cpp:66:9: warning: Return MyObj3::eventFilter() instead of false [-Wclazy-base-class-event]
